virttest/qemu_vm.py

```diff
@@ -2080,11 +2084,9 @@ def add_disable_modern(devices, dev, dev_type):
         for balloon_device in params.objects("balloon"):
             params_balloon = params.object_params(balloon_device)
             balloon_devid = params_balloon.get("balloon_dev_devid")
-            balloon_bus = None
             use_ofmt = params_balloon.get("balloon_use_old_format",
                                           "no") == "yes"
-            if params_balloon.get("balloon_dev_add_bus") == "yes":
-                balloon_bus = pci_bus
+            balloon_bus = {"aobject": params_balloon.get("balloon_bus", "pci.0")}
```

Comment:
do you need this granularity? I think simple `pci_bus` should suffice. After that you can tweak some of the actual devices to a different bus by `pci_bus_$id`. Or do you expect the frequent need to put all the ballon devices to a different bus then the hba devices or the usb controllers?